crates/iceberg/src/spec/snapshot.rs

```diff
@@ -84,24 +84,14 @@ pub struct Snapshot {
     timestamp_ms: i64,
     /// The location of a manifest list for this snapshot that
     /// tracks manifest files with additional metadata.
-    manifest_list: ManifestListLocation,
+    manifest_list: String,
```

Comment:
It would be better to add some comments that currently we only support manifest list file, and manifest files are not supported.

Reply:
Typed builder supports annotating  field with `#[builder(setter(into))]` so that we can set `manifest_list` with anything implements `Into<String>`, e.g. `&str`.

Reply:
Sorry for the confusion. What I mean is that we should still have following field definition:
```
#[builder(setter(into))]
    manifest_list: String,
```

I guess now the generated builder api for setting `manifest_list` is:

```rust
pub fn with_manifest_list(self, value: impl Into<String>) {
 ...
}
```

And we can call this api like following:
```rust
builder.with_manifest_list("test_path")
```

This way we don't need to explicitly call `"test_path".to_string()`.

Please correct me if I'm wrong.

Reply:
Ohhh, I see. I misunderstood. You mean we could add field definition to generation builder api to convert `String` to `&str` but I change the type of `manifest_list` from `String` to `&str`(with introduce lifecycle problem).
Thanks for your clear explanation! Now it looks better!